src/invoices/invoices.facade.ts

```diff
@@ -0,0 +1,50 @@
+import { IServices } from "..";
+import { IInvoiceToUpload } from "../interfaces/invoice.interface";
+import { ISenderService } from "../sender/sender.service";
+import { IFilesUploaderService } from "../uploader/uploader.service";
+import { IInvoicesService } from "./invoices.service";
+
+export interface IInvoiceFacade {
+  uploadNewInvoice: (invoiceToUpload: IInvoiceToUpload) => Promise<number>;
+}
+
+class InvoicesFacade {
```

Comment:
Mam dziwne skojarzenie ze wzorcem Fasada.

Nie jestem szczerze pewien czy ta nazwa tutaj pasuje. Wzorzec ten raczej przykrywa implementacje zewnętrznych jak freamworki/technologie (dobrym przykładem tutaj może być DateHandler -> mogła by być to fasada która zakrywa fakt że do pracy na datach korzystany z biblioteki typu dateFns/moment by nie wypływały na zewnątrz)